internal/engine/upper.go

```diff
@@ -131,40 +136,57 @@ func (u Upper) CreateManifests(ctx context.Context, manifests []model.Manifest,
 			case <-ctx.Done():
 				return ctx.Err()
 			case event := <-sw.events:
-				oldState := buildStates[event.manifest.Name]
-				buildState := oldState.NewStateWithFilesChanged(event.files)
-				buildStates[event.manifest.Name] = buildState
-
-				spurious, err := buildState.OnlySpuriousChanges()
-				if err != nil {
-					logger.Get(ctx).Infof("build watch error: %v", err)
-				}
+				if eventContainsConfigFiles(event) {
+					newManifest, err := getNewManifestFromTiltfile(ctx, event.manifest.Name)
+					if err != nil {
+						logger.Get(ctx).Infof("getting new manifest error: %v", err)
+						haveValidTiltfile = false
+						continue
+					}
+					buildState := BuildStateClean
```

Comment:
you need to store BuildStateClean in u.buildStates. otherwise, if the build fails, then previous builds will poison things